### ISSUE TITLE
summarizeService.summarize is not a function

### ISSUE THREAD

**Issue by jcfiorenzano** [clearlydefined/service#722]
requests
| where timestamp >= ago(30d)
| where resultCode == 500
| join (
exceptions
| where timestamp >= ago(30d)
) on operation_Id
| where outerMessage contains "summarizeService.summarize is not a function"

This have been happening 11 times in the last 30 days

**Comment by nellshamrell** [comment]
This query is meant to be run in the ClearlyDefined Azure Account. Application Insights -> "clearlydefined-api-prod"

Here are the API calls this happened on most recently:

* http://api.clearlydefined.io/harvest/git/github/alpinelinux/aports/b8229586a20eea5fc0f782a383ef666b6a60e054/scancode/3.2.2
* http://api.clearlydefined.io/harvest/git/github/zeit/next.js/ecf61f65a86c811efe5f2fded37889a9f2b5de96/scancode/3.2.2
* 

http://api.clearlydefined.io/harvest/git/github/alpinelinux/aports/b8229586a20eea5fc0f782a383ef666b6a60e054/scancode/3.2.2?
* 

http://api.clearlydefined.io/harvest/git/github/alpinelinux/aports/b8229586a20eea5fc0f782a383ef666b6a60e054/scancode/3.2.2
* 

http://api.clearlydefined.io/harvest/git/github/alpinelinux/aports/b8229586a20eea5fc0f782a383ef666b6a60e054/scancode/3.2.2

**Comment by nellshamrell** [comment]
The interesting thing is that all of them happened within the same 30 min (and these are the only times it happened within the past 30 days.

![image](https://user-images.githubusercontent.com/813007/105221498-0cf53b80-5b0e-11eb-95da-be9358a365f0.png)

**Comment by nellshamrell** [comment]
I ran this API call:

```bash
curl http://api.clearlydefined.io/harvest/git/github/alpinelinux/aports/b8229586a20eea5fc0f782a383ef666b6a60e054/scancode/3.2.2
```

And am not seeing the same error show up in the logs. So I think this may have been a transient infrastructure issue. Seeing what else may have been going on around that time.

**Comment by nellshamrell** [comment]
And here's the details from one of the error on 1/5/2021

```
| 0 | {"parsedStack"
  :[{"assembly":"at get (/opt/service/routes/harvest.js:26:45)",
  "method":"get","level":0,"line":26,"
  fileName":"/opt/service/routes/harvest.js"},{"assembly":"
  at processTicksAndRejections (internal/process/task_queues.js:97:5)",
 "method":"processTicksAndRejections",
  "level":1,
  "line":97,
  "fileName":"internal/process/task_queues.js"},
  {"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5",
  "method":"<no_method>",
  "level":2,"
  line":6,
  "fileName":"async /opt/service/middleware/asyncMiddleware.js"}],
  "severityLevel":"Error",
  "outerId":"0",
  "message":"summarizeService.summarize is not a function",
  "type":"TypeError",
  "id":"0"}
-- | -- | --
  | id0 messagesummarizeService.summarize is not a function 
  outerId0
  parsedStack[{"assembly":"at get (/opt/service/routes/harvest.js:26:45)",
  "method":"get",
  "level":0,"line":26,
  "fileName":"/opt/service/routes/harvest.js"},
  {"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)",
  "method":"processTicksAndRejections",
  "level":1,"
  line":97,"
  fileName":"internal/process/task_queues.js"},
  {"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5",
  "method":"<no_method>",
  "level":2,
  "line":6,
  "fileName":"async /opt/service/middleware/asyncMiddleware.js"}]0
  {"assembly":"at get (/opt/service/routes/harvest.js:26:45)",
  "method":"get","
  level":0,"
  line":26,
  "fileName":"/opt/service/routes/harvest.js"}1
  {"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)",
   "method":"processTicksAndRejections",
  "level":1,"line":97,
  "fileName":"internal/process/task_queues.js"}2
  {"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5"
  ,"method":"<no_method>",
  "level":2,
  "line":6,
  "fileName":"async /opt/service/middleware/asyncMiddleware.js"} severityLevelError typeTypeError |   | id | 0 |   | message | summarizeService.summarize is not a function |   | outerId | 0 |   | parsedStack | [{"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"},{"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"},{"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"}] |   | 0{"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"}1{"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"}2{"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"} |   | 0 | {"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"} |   | 1 | {"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"} |   | 2 | {"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"} |   | severityLevel | Error |   | type | TypeError
  | id | 0
  | message | summarizeService.summarize is not a function
  | outerId | 0
  | parsedStack | [{"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"},{"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"},{"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"}]
  | 0{"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"}1{"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"}2{"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"} |   | 0 | {"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"} |   | 1 | {"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"} |   | 2 | {"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"}
  | 0 | {"assembly":"at get (/opt/service/routes/harvest.js:26:45)","method":"get","level":0,"line":26,"fileName":"/opt/service/routes/harvest.js"}
  | 1 | {"assembly":"at processTicksAndRejections (internal/process/task_queues.js:97:5)","method":"processTicksAndRejections","level":1,"line":97,"fileName":"internal/process/task_queues.js"}
  | 2 | {"assembly":"at async /opt/service/middleware/asyncMiddleware.js:6:5","method":"<no_method>","level":2,"line":6,"fileName":"async /opt/service/middleware/asyncMiddleware.js"}
  | severityLevel | Error
  | type | TypeError
```

**Comment by nellshamrell** [comment]
I spoke too soon - looks like my request did in fact generate the error, it just took a bit to show up in the logs.

```bash
curl http://api.clearlydefined.io/harvest/git/github/alpinelinux/aports/b8229586a20eea5fc0f782a383ef666b6a60e054/scancode/3.2.2
```

![image](https://user-images.githubusercontent.com/813007/105223384-8db53700-5b10-11eb-9095-519849492150.png)

So this is reproducible.

**Comment by nellshamrell** [comment]
This was fixed with #888
